js/privatebin.js

```diff
@@ -2295,6 +2129,192 @@ jQuery.PrivateBin = function($, sjcl, Base64, RawDeflate) {
+                AttachmentViewer.hideAttachment();
+                AttachmentViewer.hideAttachmentPreview();
+                Alert.showError(I18n._('Your browser does not support uploading encrypted files. Please use a newer browser.'));
+                return;
+            }
+
+            var fileReader = new FileReader();
+            if (file === undefined) {
+                file = me.$fileInput[0].files[0];
+                $('#dragAndDropFileName').text('');
+            } else {
+                $('#dragAndDropFileName').text(file.name);
+            }
+
+            me.file = file;
+
+            fileReader.onload = function (event) {
+                var dataURL = event.target.result;
+                me.attachmentData = dataURL;
+
+                if (Editor.isPreview()) {
+                    me.handleAttachmentPreview(AttachmentViewer.$attachmentPreview, dataURL);
+                }
+            };
+            fileReader.readAsDataURL(file);
+        };
+
+        /**
+         *  Handle the preview of files that can either be an image, video, audio or pdf element.
+         *  @argument {DOM Element} targetElement where the preview should be appended.
+         *  @argument {File Data} data of the file to be displayed.
+         */
+        me.handleAttachmentPreview = function (targetElement, data) {
+            if (data) {
+                var mimeType = this.getMimeTypeFromDataURL(data);
+
+                if (mimeType.match(/image\//i)) {
+                    this.showImagePreview(targetElement, data);
+                } else if (mimeType.match(/video\//i)) {
+                    this.showVideoPreview(targetElement, data, mimeType);
+                } else if (mimeType.match(/audio\//i)) {
+                    this.showAudioPreview(targetElement, data, mimeType);
+                } else if (mimeType.match(/\/pdf/i)) {
+                    this.showPDFPreview(targetElement, data);
+                }
+                //else {
+                //console.log("file but no image/video/audio/pdf");
+                //}
+            }
+        };
+
+        /**
+         * Get Mime Type from a DataURL
+         *
+         * @param {type} dataURL
+         * @returns Mime Type from a dataURL as obtained for a file using the FileReader API https://developer.mozilla.org/en-US/docs/Web/API/FileReader#readAsDataURL()
+         */
+        me.getMimeTypeFromDataURL = function (dataURL) {
+            return dataURL.slice(dataURL.indexOf('data:') + 5, dataURL.indexOf(';base64,'));
+        };
+
+        me.showImagePreview = function (targetElement, image) {
+            targetElement.html(
+                    $(document.createElement('img'))
+                    .attr('src', image)
+                    .attr('class', 'img-thumbnail')
+                    );
+            targetElement.removeClass('hidden');
+        };
+
+        me.showVideoPreview = function (targetElement, video, mimeType) {
```

Comment:
As targetElement is a jQuery element prepend the variable with `$`. So: `$targetElement`.
Also JSDOC would be nice.